server/apps/immich/src/api-v1/asset/asset.controller.ts

```diff
@@ -77,13 +77,19 @@ export class AssetController {
     @UploadedFiles() files: { assetData: Express.Multer.File[]; livePhotoData?: Express.Multer.File[] },
     @Body(ValidationPipe) createAssetDto: CreateAssetDto,
     @Response({ passthrough: true }) res: Res,
+    @Query('key') sharedKey?: string,
   ): Promise<AssetFileUploadResponseDto> {
+    if (sharedKey) {
+      await this.assetService.checkUploadAccess(sharedKey, files.assetData[0].path);
```

Comment:
This upload check should happen before the file is uploaded, which would prevent needing to delete it.